src/R/Components/Impl/History/Implementation/RHistory.cs

```diff
@@ -364,32 +373,41 @@ internal sealed class RHistory : IRHistory {
                 return;
             }
 
+            IRHistoryEntry entryToSelect;
             if (HasSelectedEntries) {
                 var firstSelectedEntry = _entries.GetSelectedEntries().First();
                 if (firstSelectedEntry.Previous == null) {
                     return;
                 }
 
                 ClearHistoryEntrySelection();
-                firstSelectedEntry.Previous.IsSelected = true;
+                entryToSelect = firstSelectedEntry.Previous;
             } else {
                 if (VisualComponent == null) {
                     return;
                 }
 
                 if (VisualComponent.TextView.Selection.IsEmpty) {
-                    _entries.LastOrDefault().IsSelected = true;
+                    entryToSelect = _entries.LastOrDefault();
                 } else {
                     var entry = GetHistoryEntryFromPosition(VisualComponent.TextView.Selection.Start.Position);
                     if (entry.Previous == null) {
                         return;
                     }
 
                     VisualComponent.TextView.Selection.Clear();
-                    entry.Previous.IsSelected = true;
+                    entryToSelect = entry.Previous;
                 }
             }
 
+            entryToSelect.IsSelected = true;
```

Comment:
`_entries.Count` shouldn't be `0` because `HasEntries` is true. I'll replace it with `Last()`